supermarq-benchmarks/supermarq/qcvv/base_experiment.py

```diff
@@ -508,7 +508,17 @@ def _interleave_op(
         operation = operation.with_tags("no_compile")
         interleaved_circuit = circuit.copy()
         interleaved_circuit.batch_insert(
-            [(k, operation) for k in range(len(circuit) - int(not include_final), 0, -1)]
+            [
+                (
+                    k,
+                    [
+                        css.barrier(*interleaved_circuit.all_qubits()),
+                        operation,
+                        css.barrier(*interleaved_circuit.all_qubits()),
```

Comment:
```suggestion
                        css.barrier(*self.qubits),
                        operation,
                        css.barrier(*self.qubits),
```

Reply:
(sorry just realized this also requires making `_interleave_op` not a staticmethod (which imo it shouldn't be anyway))

Reply:
interesting, yeah maybe (thought ideally it would be nice if we could make the same method work for more protocols)

probably not for this pr though - would love to get this in so i can build on it :)